autorest/azure/environments.go

```diff
@@ -0,0 +1,99 @@
+package azure
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+)
+
+// Environment represents a set of endpoints for each of Azure's Clouds.
+type Environment struct {
+	Name                      string
+	ManagementPortalURL       string
+	PublishSettingsURL        string
+	ServiceManagementEndpoint string
+	ResourceManagerEndpoint   string
+	ActiveDirectoryEndpoint   string
+	GalleryEndpoint           string
+	KeyVaultEndpoint          string
+	GraphEndpoint             string
+	StorageEndpointSuffix     string
+	SQLDatabaseDNSSuffix      string
+	TrafficManagerDNSSuffix   string
+	KeyVaultDNSSuffix         string
+	ServiceBusEndpointSuffix  string
+}
+
+var (
+	// PublicCloud is the default public Azure cloud environment
+	PublicCloud = Environment{
+		Name:                      "AzurePublicCloud",
+		ManagementPortalURL:       "https://manage.windowsazure.com/",
+		PublishSettingsURL:        "https://manage.windowsazure.com/publishsettings/index",
+		ServiceManagementEndpoint: "https://management.core.windows.net/",
+		ResourceManagerEndpoint:   "https://management.azure.com/",
+		ActiveDirectoryEndpoint:   "https://login.microsoftonline.com/?api-version=1.0",
```

Comment:
I think the `api-version` should not be in this string as we never actually query https://login.microsoftonline.com/?api-version=1.0 so it makes sense to extract api-version into a separate const.